api/porchconfig/v1alpha1/types.go

```diff
@@ -92,6 +92,8 @@ type GitRepository struct {
 	// Address of the Git repository, for example:
 	//   `https://github.com/GoogleCloudPlatform/blueprints.git`
 	Repo string `json:"repo"`
+	// +kubebuilder:default=main
+	// +kubebuilder:validation:MinLength=1
```

Comment:
How does this get enforced? At k8s level?

Reply:
yes, this is enforced by the main Kubernetes API server